auto_submit/lib/service/pull_request_validation_service.dart

```diff
@@ -130,6 +142,20 @@ class PullRequestValidationService extends ValidationService {
       await githubService.createComment(slug, prNumber, message);
       log.info(message);
     } else {
+      // Remove the autosubmit label post enqueue/merge to avoid infinite loops.
+      // Here's an example of an infinite loop:
+      //
+      // 1. Autosubmit bot is notified that a PR is ready (reviewed, all green, has `autosubmit` label).
+      // 2. Autosubmit bot puts the PR onto the merge queue.
+      // 3. The PR fails some tests in the merge queue.
+      // 4. Github kicks the PR back, removing it fom the merge queue.
+      // 5. GOTO step 1.
+      //
+      // Removing the `autosubmit` label will prevent the autosubmit bot from
```

Comment:
This sounds perfectly fine to me. Maybe we come up with more labels in the future to better describe / search what was happening ("autosubmit-attempted", "mq-failed previously") but that's MVP+1

Reply:
The default Github UI actually provides a pretty detailed audit trail. Take a look at this PR that has endured a lot of different situations:

https://github.com/flutter/flaux/pull/42